writers/influxdb/influxdb.go

```diff
@@ -0,0 +1,76 @@
+package influxdb
+
+import (
+	"strconv"
+	"time"
+
+	client "github.com/influxdata/influxdb/client/v2"
+	"github.com/mainflux/mainflux"
+)
+
+var _ Writer = (*influxRepo)(nil)
+
+type influxRepo struct {
+	database  string
+	pointName string
+	client    client.Client
+}
+
+type fields map[string]interface{}
+type tags map[string]string
+
+func convertMsg(msg mainflux.Message) (tags, fields) {
+	time := strconv.FormatFloat(msg.Time, 'f', -1, 64)
+	update := strconv.FormatFloat(msg.UpdateTime, 'f', -1, 64)
+	tags := map[string]string{
+		"Channel":    msg.Channel,
+		"Publisher":  msg.Publisher,
+		"Protocol":   msg.Protocol,
+		"Name":       msg.Name,
+		"Unit":       msg.Unit,
+		"Link":       msg.Link,
+		"Time":       time,
+		"UpdateTime": update,
+	}
+
+	fields := map[string]interface{}{
+		"Value":       msg.Value,
+		"ValueSum":    msg.ValueSum,
+		"BoolValue":   msg.BoolValue,
+		"StringValue": msg.StringValue,
+		"DataValue":   msg.DataValue,
+	}
+	return tags, fields
+}
+
+// New returns new InfluxDB writer.
+func New(cfg client.HTTPConfig, database, pointName string) (Writer, error) {
```

Comment:
Instead of passing the configuration, instantiate the client inside `main.go` and do the `defer client.Close()` there. This would remove the entire `Writer` interface and let you rely only on `MessageRepository` as agreed.